lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -1944,6 +1945,7 @@ buildSignSubmitTransaction db@DBLayer{..} netLayer txLayer pwd walletId
 buildAndSignTransactionPure
     :: forall k s n
      . ( WalletKey k
+       , TxWitnessTagFor k
```

Comment:
If you want to avoid adding these constraints you could probably re-retrieve them from `TransactionLayer` in some way, e.g. adding a `txWitnessTag` field to `TransactionLayer`.